### PR TITLE
convert bytes object to string

### PR DESCRIPTION
**Pull request by migueldvb**
This works with python 3 and 2.7

### CODE REVIEW

**Comment by gawrysz** [comment]
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe8' in position 2736: ordinal not in range(128)

The error occurs in src/base/units.F90 in line 55 on 'è' placed in a comment.

**Comment by migueldvb** [comment]
I have moved the decode to the problems/*/info ptint statement and the chacks pass.

**Comment by Xarthisius** [comment]
Bytes vs string will apply to every call of `get_stdout`. Instead of fixing each invocation you could modify the function itself. There at least two `try: ... except UnicodeError: ...` which also mask the same issue. They should be removed. You can check for python version inside `get_stdout` using: `if sys.version_info >= (3,0,0):`

**Comment by migueldvb** [comment]
That is a good idea. I have modifies the get_stdout function and removed the `except UnicodeError1 in the setup script.

File: python/piernik_setup.py [hunk 1]
```diff
@@ -332,11 +335,10 @@ def setup_piernik(data=None):
     cmd = "echo '#include \"%spiernik.h\"' > \"%s\"" % ('src/base/', foo_path)
     cmd += " && cpp %s -dM -I%s \"%s\" && rm \"%s\"" % (
         cppflags, probdir, foo_path, foo_path)
-    try:
+    if sys.version_info >= (3,0,0):
         defines = sp.Popen([cmd], stdout=sp.PIPE,
```
Comment (Xarthisius) [inline python/piernik_setup.py]: You can use `get_stdout` here too.

`defines = get_stdout(cmd).rstrip().split("\n")`

**Comment by migueldvb** [comment]
Fixed.

**Comment by migueldvb** [comment]
Still some test is failing...

**Comment by gawrysz** [comment]
Jenkins test this please

**Pull request merged**